pkg/backupdriver/backup_driver_controller_base.go

```diff
@@ -396,6 +396,17 @@ func (ctrl *backupDriverController) syncSnapshotByKey(key string) error {
 
 	if snapshot.Status.Phase != backupdriverapi.SnapshotPhaseNew {
 		ctrl.logger.Debugf("Skipping snapshot, %v, which is not in New phase. Current phase: %v", key, snapshot.Status.Phase)
+		if snapshot.Status.Phase == backupdriverapi.SnapshotPhaseUploaded {
+			// If Snapshot CR status reaches terminal state more than 1 hour, Snapshot CR should be deleted
+			now := time.Now()
+			if snapshot.Status.CompletionTimestamp.Add(time.Hour).Unix() < now.Unix() {
```

Comment:
I don't think this 1 hour should be hard-coded.  It should be configurable.  Also default of 1 hour is too short.

Reply:
Change to use configurable time period. This change uses the same time constant as https://github.com/vmware-tanzu/velero-plugin-for-vsphere/pull/186, will merge PR 186 first and resolve the conflicts.